src/NzbDrone.Core/Notifications/Slack/SlackSettings.cs

```diff
@@ -0,0 +1,36 @@
+﻿using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.Notifications.Slack
+{
+    public class SlackSettingsValidator : AbstractValidator<SlackSettings>
+    {
+        public SlackSettingsValidator()
+        {
+            RuleFor(c => c.WebHookUrl).IsValidUrl();
+            RuleFor(c => c.BotName).NotEmpty();
+            RuleFor(c => c.Icon).NotEmpty();
+        }
+    }
+
+    public class SlackSettings : IProviderConfig
+    {
+        private static readonly SlackSettingsValidator Validator = new SlackSettingsValidator();
+
+        [FieldDefinition(0, Label = "WebHookUrl", HelpText = "slack channel webhook url.", Type = FieldType.Url, HelpLink = "https://my.slack.com/services/new/incoming-webhook/")]
```

Comment:
The label is intended to be a human readable string, `Webhook URL` is better (same thing applies to the other labels).